builtin/mainmenu/dlg_contentstore.lua

```diff
@@ -123,6 +139,290 @@ local function start_install(calling_dialog, package)
 	end
 end
 
+local function queue_download(package)
+	local max_concurrent_downloads = tonumber(minetest.settings:get("contentdb_max_concurrent_downloads"))
+	if number_downloading < max_concurrent_downloads then
+		start_install(package)
+	else
+		table.insert(download_queue, package)
+		package.queued = true
+	end
+end
+
+local function get_raw_dependencies(package)
+	if package.raw_deps then
+		return package.raw_deps
+	end
+
+	print("#### Fetching raw deps for " .. package.id)
+
+	local tmpdir = os.tempfolder()
+	local target = tmpdir .. DIR_DELIM .. "package_deps.json"
+
+	assert(core.create_dir(tmpdir))
+
+	local url_fmt = "/api/packages/%s/dependencies/?protocol_version=%s&engine_version=%s&only_hard=1"
+	local version = core.get_version()
+	local base_url = core.settings:get("contentdb_url")
+	local url = base_url .. url_fmt:format(package.id, core.get_max_supp_proto(), version.string)
+	core.download_file(url, target)
```

Comment:
does the engine not have a method to download an url into a string?

Reply:
It does not in the menu API, but does in the server-side API. Perhaps I shou I could add it to the menu API?